src/main/java/com/maxmind/db/Reader.java

```diff
@@ -222,20 +279,98 @@ private int findIpV4StartNode(ByteBuffer buffer)
         return node;
     }
 
-    private int readNode(ByteBuffer buffer, int nodeNumber, int index)
-        throws InvalidDatabaseException {
+    /**
+     * Returns an iterator within the specified network.
+     * Please note that a MaxMind DB may map IPv4 networks into several locations
+     * in an IPv6 database. This iterator will iterate over all of these locations
+     * separately. To only iterate over the IPv4 networks once, use the
+     * SkipAliasedNetworks option.
+     * @param <T> Represents the data type(e.g., Map, HastMap, etc.).
+     * @param network Specifies the network to be iterated.
+     * @param skipAliasedNetworks Boolean for skipping aliased networks.
+     * @return Networks
+     * @throws BadVersionException Exception for using an IPv6 network in ipv4-only database.
+     * @throws ClosedDatabaseException Exception for a closed databased.
+     * @throws InvalidDatabaseException Exception for an invalid database.
+     */
+    public <T> Networks<T> networksWithIn(Network network, boolean skipAliasedNetworks) 
+        throws BadVersionException, ClosedDatabaseException, InvalidDatabaseException {
+        InetAddress networkAddress = network.getNetworkAddress();
+        if (this.metadata.getIpVersion() == 4 && networkAddress instanceof Inet6Address) {
+            throw new BadVersionException(networkAddress);
+        }
+
+        byte[] ipBytes = networkAddress.getAddress();
+        int prefixLength = network.getPrefixLength();
+
+        if (this.metadata.getIpVersion() == 6 && ipBytes.length == IPV4_LEN) {
+            if (skipAliasedNetworks) {
+                ipBytes = new byte[]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                        ipBytes[0], ipBytes[1], ipBytes[2], ipBytes[3] };
+            } else {
+                // Convert it to the IP address (in 16-byte from) of the IPv4 address.
```

Comment:
I am not sure if we should use this version as it is invalid when aliasing is disabled when the database is created. The IPv4 subtree is defined in the format as existing at `::/96`. Perhaps we could just set `skipAliasedNetworks` to true if they pass in an IPv4 address and we have an IPv6 tree.

Reply:
I made it the same as what the go reader does by calling [IPv6](https://github.com/oschwald/maxminddb-golang/blob/eeae70400a52e32fb292421afbf198ad08e9005f/traverse.go#L92). 

So, are you saying that I should set `ipBytes` to
```
ipBytes = new byte[]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                        ipBytes[0], ipBytes[1], ipBytes[2], ipBytes[3] };
```
every time `if (this.metadata.getIpVersion() == 6 && ipBytes.length == IPV4_LEN) {` is true ? So, ipv4 will always go to `::/96` ?

Reply:
The Go library has to deal with the weird `net.IP` type and I wouldn't necessarily follow it exactly here. I think it would be better to treat `::/96` as the canonical location of the IPv4 data as it is in the specification.

I assume the test is failing due to [this code](https://github.com/maxmind/MaxMind-DB-Reader-java/pull/131/files#diff-a34fba8a63b9f08b67dd9f7e79c0bf86bbc58e4c678d651d2143573a9dee2727R90). If you remove the skip check there, I think this should work.